libcudacxx/include/cuda/std/detail/libcxx/include/span

```diff
@@ -257,6 +262,40 @@ _LIBCUDACXX_INLINE_VAR constexpr bool __is_span_compatible_container<
                 nullptr_t>>> = true;
 #  endif // _CCCL_STD_VER <= 2014 || _CCCL_COMPILER_MSVC_2017
 
+#  if _CCCL_STD_VER >= 2020
+
+template <class _Tp>
+concept __integral_constant_like =
+  is_integral_v<decltype(_Tp::value)> //
+  && !is_same_v<bool, remove_const_t<decltype(_Tp::value)>> //
+  && convertible_to<_Tp, decltype(_Tp::value)> //
+  && equality_comparable_with<_Tp, decltype(_Tp::value)> //
+  && bool_constant<_Tp() == _Tp::value>::value
```

Comment:
The difference is that this requires `_Tp() == _Tp::value` to be a compile time constant

Reply:
So does my version, if I read the section on `Atomic constraints` [cppreference](https://en.cppreference.com/w/cpp/language/constraints):

> Otherwise, E, after any lvalue-to-rvalue conversion, shall be a prvalue constant expression of type bool, and the constraint is satisfied if and only if it evaluates to true.

Where E is an expression such as `_Tp() == _Tp::value`. So I would argue my version is equivalent.

Reply:
(btw, no PR changes are necessary! I would prefer if you stick to the wording, where they use `bool_constant`)

Reply:
Its a bit conceived, but something that would for example not define a proper constexpr comparison operator would fail:
https://godbolt.org/z/sesdj3rE3

Reply:
@bernhardmgruber @miscco Hi!  I didn't see this notification last week.  The reason for using `bool_constant` in the definition of _`integral-constant-like`_ is to make sure that `T() == T::value` returns something convertible to `bool`.  On the other hand, `equality_comparable_with<T, decltype(T::value)>` probably implies that anyway, so `bool_constant` might be superfluous there.  It's likely that any redundancy in the wording is related to the implementation that needs to back-port to C++17.